File: backend/app/routers/contracts.py
from fastapi import APIRouter, HTTPException, Depends
from app.database import get_supabase
from app.dependencies import verify_token

router = APIRouter(prefix="/api/contracts", tags=["contracts"])


@router.get("/{contract_id}")
async def get_contract(contract_id: str, auth: dict = Depends(verify_token)):
    supabase = get_supabase()
    result = supabase.table("contracts").select("*").eq("id", contract_id).execute()
    if not result.data:
        raise HTTPException(status_code=404, detail="Contract not found")
    return {"success": True, "data": result.data[0]}

File: backend/src/routes/contracts.ts
import { Router, Request, Response } from "express";
import { verifyToken } from "../middleware/auth";
import { supabase } from "../database";

const router = Router();

router.get("/:id", verifyToken, async (req: Request, res: Response) => {
  try {
    const { data, error } = await supabase
      .from("contracts")
      .select("*")
      .eq("id", req.params.id);
    if (error) return res.status(400).json({ detail: error.message });
    if (!data || data.length === 0) {
      return res.status(404).json({ detail: "Contract not found" });
    }
    return res.json({ success: true, data: data[0] });
  } catch (err: any) {
    return res.status(500).json({ detail: err.message });
  }
});

export default router;
